source/llassetgen/include/llassetgen/DistanceTransform.h

```diff
@@ -63,6 +63,8 @@ namespace llassetgen {
         std::unique_ptr<Parabola[]> parabolas;
         std::unique_ptr<OutputType[]> lineBuffer;
 
+        template<bool fill>
+        void edgeDetection(DimensionType offset, DimensionType pitch, DimensionType length);
```

Comment:
Usually the source code of a template function is provided with the header (we use `.inl` files that are included at the end of the header file). Since this is a private function, this will probably be no problem.